db/compaction/compaction_iterator_test.cc

```diff
@@ -1033,6 +1055,194 @@ INSTANTIATE_TEST_CASE_P(CompactionIteratorWithAllowIngestBehindTestInstance,
                         CompactionIteratorWithAllowIngestBehindTest,
                         testing::Values(true, false));
 
+class CompactionIteratorTsGcTest : public CompactionIteratorTest {
+ public:
+  CompactionIteratorTsGcTest()
+      : CompactionIteratorTest(test::ComparatorWithU64Ts()) {}
+};
+
+TEST_P(CompactionIteratorTsGcTest, NoKeyEligibleForGC) {
+  constexpr char user_key[] = "a";
+  const std::vector<std::string> input_keys = {
+      test::KeyStr(/*ts=*/103, user_key, /*seq=*/4, kTypeValue),
+      test::KeyStr(/*ts=*/102, user_key, /*seq=*/3,
+                   kTypeDeletionWithTimestamp)};
+  const std::vector<std::string> input_values = {"a3", "a2"};
```

Comment:
I'm a bit confused here; does `a2` here belong to the KV with `kTypeDeletionWithTimestamp`? Are you using these as dummy values to be able to check whether certain tombstones got preserved? (Applies to other test cases below as well.)

Reply:
Yes. "a2" is just a dummy value.

Reply:
Apparently the prevailing convention is to simply use an empty string for deletions; we could consider doing that.